lib/api/BackbeatAPI.js

```diff
@@ -98,79 +98,105 @@ class BackbeatAPI {
     }
 
     /**
-     * Check if incoming request is valid
+     * Find valid route from list of routes defined in Arsenal backbeat routes
+     * Sets the matched route on the BackbeatRequest object with possible
+     * extra properties.
      * @param {BackbeatRequest} bbRequest - holds relevant data about request
-     * @return {boolean} true/false
+     * @return {Object|null} - The error object or `null` if no error
      */
-    isValidRoute(bbRequest) {
+    findValidRoute(bbRequest) {
+        /* eslint-disable no-param-reassign */
         const rDetails = bbRequest.getRouteDetails();
-        if (!rDetails) {
-            return false;
-        }
         const route = bbRequest.getRoute();
-        // first validate healthcheck routes
+        const addKeys = {};
+
+        // first validate healthcheck routes or prom routes since they do not
+        // have rDetails set
         if (route === 'healthcheck') {
-            return true;
+            if (bbRequest.getHTTPVerb() !== 'GET') {
+                return errors.MethodNotAllowed
+                    .customizeDescription('invalid http verb');
+            }
+            // TODO: this logic changes with addition of ddep healthcheck
+            const healthcheckRoute = routes.filter(r =>
+                r.category === 'healthcheck')[0];
+            bbRequest.setMatchedRoute(healthcheckRoute);
+            return null;
         }
         if (route === 'monitoring/metrics') {
-            return true;
+            if (bbRequest.getHTTPVerb() !== 'GET') {
+                return errors.MethodNotAllowed
+                    .customizeDescription('invalid http verb');
+            }
+            const promRoute = routes.filter(r =>
+                r.category === 'monitoring')[0];
+            bbRequest.setMatchedRoute(promRoute);
+            return null;
         }
 
-        /*
-            {
-                category: 'metrics',
-                extension: 'crr',
-                site: 'my-site-name',
-                metric: 'backlog', (optional)
-            }
-        */
-        // check metric routes
-        // Are there any routes with matching extension?
-        const extensions = routes.reduce((store, r) => {
-            if (r.extensions[rDetails.extension] &&
-                r.extensions[rDetails.extension].includes(rDetails.status)) {
-                store.push(Object.keys(r.extensions));
-            } else if (rDetails.category === 'metrics') {
-                store.push(Object.keys(r.extensions));
+        // All routes at this point have extensions property in use
+        // Match http verb and match extension name
+        let filteredRoutes = routes.filter(r => (
+            bbRequest.getHTTPVerb() === r.httpMethod &&
+            Object.keys(r.extensions).includes(rDetails.extension)
```

Comment:
A more straightforward and efficient way is to write `&& r.extensions[rDetails.extension]`